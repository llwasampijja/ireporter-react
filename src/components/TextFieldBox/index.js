import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const TextFieldBox = (props) => {
  const {
    textFieldName,
    textFieldValue,
    textFieldId,
    textFieldType,
    textFieldClass,
    textFieldOnchange,
    textFieldPlaceholder,
    children,
  } = props;

  return (
    <Fragment>
      <input
        id={textFieldId}
        name={textFieldName}
        value={textFieldValue}
        type={textFieldType}
        className={textFieldClass}
        onChange={textFieldOnchange}
        placeholder={textFieldPlaceholder}
      />
      {children}
    </Fragment>
  );
};

TextFieldBox.propTypes = {
  textFieldName: PropTypes.string,
  textFieldValue: PropTypes.string,
  textFieldId: PropTypes.string,
  textFieldType: PropTypes.string,
  textFieldClass: PropTypes.string,
  textFieldOnchange: PropTypes.func,
  textFieldPlaceholder: PropTypes.string,
  children: PropTypes.node,
};

TextFieldBox.defaultProps = {
  textFieldName: '',
  textFieldValue: '',
  textFieldId: '',
  textFieldType: '',
  textFieldClass: '',
  textFieldOnchange: () => {},
  textFieldPlaceholder: '',
  children: '',
};

export default TextFieldBox;
