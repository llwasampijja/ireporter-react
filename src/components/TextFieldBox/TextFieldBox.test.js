// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import TextFieldBox from '.';

describe('tests the TextFieldBox component', () => {
  const wrapper = shallow(<TextFieldBox />);
  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
