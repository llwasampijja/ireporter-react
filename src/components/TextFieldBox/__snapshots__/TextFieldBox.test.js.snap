// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the TextFieldBox component should match snapshop 1`] = `
<Fragment>
  <textarea
    className=""
    id=""
    name=""
    onChange={[Function]}
    placeholder=""
    type=""
    value=""
  />
</Fragment>
`;
